core/trino-main/src/main/java/io/trino/metadata/DiscoveryNodeManager.java

```diff
@@ -215,9 +220,12 @@ private synchronized void refreshNodesInternal()
                 .filter(service -> !failureDetector.getFailed().contains(service))
                 .collect(toImmutableSet());
 
-        ImmutableSet.Builder<InternalNode> activeNodesBuilder = ImmutableSet.builder();
-        ImmutableSet.Builder<InternalNode> inactiveNodesBuilder = ImmutableSet.builder();
-        ImmutableSet.Builder<InternalNode> shuttingDownNodesBuilder = ImmutableSet.builder();
+        // Map from NodeState to ImmutableSet.Builder of InternalNode.
```

Comment:
remove, it is not more helpful than varialble type

Reply:
Also it looks like it would be cleaner if you just used ` ImmutableSetMultimap.Builder<NodeState, InternalNode> `

Reply:
done